ocs_ci/framework/__init__.py

```diff
@@ -352,5 +352,59 @@ def is_consumer_exist(self):
         cluster_types = [cluster.ENV_DATA["cluster_type"] for cluster in self.clusters]
         return "consumer" in cluster_types
 
+    def is_cluster_type_exist(self, cluster_type):
+        """
+        Check if the given cluster type exists in the clusters
+
+        Args:
+            cluster_type (str): The cluster type
+
+        Returns:
+            bool: True, if the given cluster type exists in the clusters. False, otherwise.
+
+        """
+        cluster_types = [cluster.ENV_DATA["cluster_type"] for cluster in self.clusters]
+        return cluster_type in cluster_types
+
+    def get_cluster_type_indexes_list(self, cluster_type):
+        """
+        Get the cluster type indexes
+
+        Returns:
+            list: the cluster type indexes
+
+        Raises:
+            ClusterNotFoundException: In case it didn't find any cluster with the cluster type
+
+        """
+        cluster_type_indexes_list = []
+        for i, cluster in enumerate(self.clusters):
+            if cluster.ENV_DATA["cluster_type"] == cluster_type:
+                cluster_type_indexes_list.append(i)
+
+        if not cluster_type_indexes_list:
+            raise ClusterNotFoundException(
+                f"Didn't find any cluster with the cluster type '{cluster_type}'"
+            )
+
+        return cluster_type_indexes_list
+
+    def switch_to_cluster_by_cluster_type(self, cluster_type, num_of_cluster=0):
+        """
+        Switch to the cluster with the given cluster type
+
+        Args:
+            cluster_type (str): The cluster type
+            num_of_cluster (int): The cluster index to switch to. The default cluster number
+                is 0 - which means it will switch to the first cluster.
+                1 - is the second, 2 - is the third, and so on.
+        Raises:
+            ClusterNotFoundException: In case it didn't find any cluster with the cluster type
+
+        """
+        self.switch_ctx(
+            self.get_cluster_type_indexes_list(cluster_type)[num_of_cluster]
```

Comment:
IIUC, `get_cluster_type_indexes_list()` returns a list of indices in a form of integers, e.g. `[2,4,5]`. 
But here you are switching context to the index in the list of integers.
So based on the above example of `[2,4,5]`, if `num_of_cluster` is 2, you will switch to cluster `5` and not `2` as `5` is the 3rd index in the list

Reply:
Yes, that should be the behavior—the same logic as the existing method `switch_to_consumer`: https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/__init__.py#L296. Maybe the docstring is not clear enough.